sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/orderbyquery/OrderbyRowComparer.java

```diff
@@ -64,7 +64,7 @@ public int compare(OrderByRowResult<T> r1, OrderByRowResult<T> r2) {
                     case Ascending:
                         return cmp;
                     case Descending:
-                        return -cmp;
+                        return ItemComparator.getInstance().compare(result2.get(i).getItem(), result1.get(i).getItem());
```

Comment:
This recomputes the cmp, instead you can use signum

Reply:
Yes, I was debating the same if I should recompute or use +1/-1.

Using signum is a good idea. I've made the change. I've changed the Ascending case as well to keep the result consistent. Please let me know if the change looks good.